daffodil-runtime1/src/main/scala/org/apache/daffodil/layers/LayerTransformer.scala

```diff
@@ -216,7 +216,9 @@ class JavaIOInputStream(s: InputSourceDataInputStream, finfo: FormatInfo)
     }
   }
 
-  override def available(): Int = 1
+  override def available(): Int = {
+    Assert.usageError("Not to be called on JavaIOInputStream.")
```

Comment:
This is causing failures of unit tests.
I don't understand the prior definition of 1. That means there is always expectation that asking for 1 more byte will not hang, doesn't seem to meet the API definition of available(). 
Not sure we want to be doing a blocking read here so we can return 0 if there is no more data, or not.